src/main/java/br/com/dh/desafio_spring/service/ArticleService.java

```diff
@@ -56,6 +56,9 @@ public List<Article> findAllByFreeShippingAndPrestige(Boolean freeShipping, Stri
 
     public List<Article> findAllByCategoryAndFreeShippingOrdered(String category, Boolean freeShipping, int order) {
         List<Article> articleList = this.findAllByCategoryAndFreeShipping(category, freeShipping);
+
+        if(articleList.isEmpty()) throw new NotFoundException("Nenhum dado encontrado!");
+
```

Comment:
Oi, @SouzaGiovannaMELI !

Esta verificação já existe na função `"this.findAllByCategoryAndFreeShipping(category, freeShipping);"`, então o erro seria lançado direto nela e jamais chegaria nesse if, já que essa função está sendo chamada na linha 58, faz sentido?

> preview da função citada
```java
    @Override
    public List<Article> findAllByFreeShippingAndPrestige(Boolean freeShipping, String prestige) {
        List<Article> article = repo.findAllByFreeShippingAndPrestige(freeShipping, prestige);

        if (article.isEmpty()) {
            throw new NotFoundException("Objeto não encontrado");
        }

        return article;
    }
  ```